app/src/trackPlayer.ts

```diff
@@ -0,0 +1,78 @@
+import RNTrackPlayer, { Track, Capability, Event, State } from 'react-native-track-player'
+
+export const initialPlayerState = {
+    currentTrack: null,
+    currentIndex: null,
+    recordsCount: 0,
+    isPlaying: false
+}
+
+export async function createPlayerState(event: {
+    type: Event;
+    [key: string]: any;
+}) {
+    const queue = await RNTrackPlayer.getQueue()
+    const currentIndex = await RNTrackPlayer.getCurrentTrack()
+    const state = await RNTrackPlayer.getState();
+    const track = await RNTrackPlayer.getTrack(currentIndex)
+
+    return {
+        currentTrack: track,
+        currentIndex: currentIndex,
+        recordsCount: queue.length,
+        isPlaying: state == State.Playing
+    }
+}
+
+class TrackPlayer {
+    private static instance: TrackPlayer
+
+    static getInstance() {
+        if (!TrackPlayer.instance) {
+            TrackPlayer.instance = new TrackPlayer()
+            TrackPlayer.instance.init()
+            return TrackPlayer.instance
+        }
+
+        return TrackPlayer.instance
+    }
+
+    private init() {
+        RNTrackPlayer.setupPlayer({})
+
+        const capabilities = [
+            Capability.Play,
+            Capability.Pause,
+            Capability.SeekTo
+        ]
+        const options = {
+            stopWithApp: false,
+            capabilities,
+            compactCapabilities: capabilities
+        }
+        RNTrackPlayer.updateOptions(options)
```

Comment:
drobnost, ale definoval bych ty options primo v `updateOptions`, je tam lepsi TS support (byt teda i takto to validuje, ale chyby jsou hure citelne)